client/src/components/app.test.js

```diff
@@ -1,11 +1,21 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import App from './app';
+import { BrowserRouter as Router } from 'react-router-dom';
+import store from '../redux/store';
+import { Provider } from 'react-redux';
 
 describe('Application', function() {
   it('renders without crashing', () => {
     const div = document.createElement('div');
 
-    ReactDOM.render(<App />, div);
+    ReactDOM.render(
+      <Provider store={store}>
+        <Router>
+          <App />
+        </Router>
+      </Provider>,
+      div
+    );
```

Comment:
This took a bit of search... so part of the issue with these components is we have an async aspect, hence one of the problems with Fetch (not the non-existent ```finally``` method). Adding this ```React.unmountComponentAtNode(div)``` after line 19 ensures the initial mount and immediate unmount of the component for a quick test, soooo like so:

```js
ReactDOM.render(
  <Provider store={store}>
    <Router>
      <App />
    </Router>
  </Provider>,
  div
);

ReactDOM.unmountComponentAtNode(div);
```
If we combine this with converting the ```finally``` to a ```then``` or simply removing them. The tests should all pass with just a general console warning about the redirect... which we can address later.

Found this here... https://github.com/facebook/create-react-app/issues/3482#issuecomment-347347454